sdc/numpy/sdc_numpy_modified_functions.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+# *****************************************************************************
+# Copyright (c) 2020, Intel Corporation All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+"""
+
+| This file contains SDC numpy modified functions, rewritten according to concurrency
+
+"""
+
+import numba
+import numpy
+
+from numba.errors import TypingError
+from numba import types
+from numba import jit
+from numba import prange
+from numba import numpy_support
+
+import sdc
+from sdc.datatypes.common_functions import TypeChecker
+from sdc.str_arr_ext import (StringArrayType, pre_alloc_string_array)
+from sdc.utils import sdc_overload
+
+
+def astype(self, dtype, order='K', casting='unsafe', subok=True, copy=True):
+    pass
+
+
+@sdc_overload(astype, jit_options={'parallel': True})
+def sdc_astype_overload(self, dtype, order='K', casting='unsafe', subok=True, copy=True):
```

Comment:
We don't need to completely reproduce numpy interface by the way

Reply:
Yes. It is possible to use `@numba.extending.register_jitable` for creating reused functions.

Reply:
We need some kind of `sdc_register_jitable`. Also `register_jitable` doesn't care about types, and this is not what we want from `astype`. Our `astype` should do nothing if input and target types are the same